evennia/contrib/game_systems/containers/containers.py

```diff
@@ -0,0 +1,301 @@
+            singular, _ = self.get_numbered_name(1, putter)
+            putter.msg(f"You can't fit anything else in {singular}.")
+            return False
+
+        return True
+
+
+class CmdContainerLook(CmdLook):
+    """
+    look at location or object
+
+    Usage:
+      look
+      look <obj>
+      look <obj> in <container>
+      look *<account>
+
+    Observes your location or objects in your vicinity.
+    """
+
+    rhs_split = (" in ",)
+
+    def func(self):
+        """
+        Handle the looking.
+        """
+        caller = self.caller
+        # by default, we don't look in anything
+        container = None
+
+        if not self.args:
+            target = caller.location
+            if not target:
+                self.msg("You have no location to look at!")
+                return
+        elif self.rhs:
+            # we are looking in something, find that first
+            container = caller.search(self.rhs)
+            if not container:
+                return
+
+        target = caller.search(self.lhs, location=container)
+        if not target:
+            return
+
+        desc = caller.at_look(target)
+        # add the type=look to the outputfunc to make it
+        # easy to separate this output in client.
+        self.msg(text=(desc, {"type": "look"}), options=None)
+
+
+class CmdContainerGet(CmdGet):
+    """
+    pick up something
+
+    Usage:
+      get <obj>
+      get <obj> from <container>
+
+    Picks up an object from your location or a container and puts it in
+    your inventory.
+    """
+
+    rhs_split = (" from ",)
+
+    def func(self):
+        caller = self.caller
+        # by default, we get from the caller's location
+        location = caller.location
+
+        if not self.args:
+            self.msg("Get what?")
+            return
+
+        # check for a container as the location to get from
+        if self.rhs:
+            location = caller.search(self.rhs)
+            if not location:
+                return
+            # check access lock
+            if not location.access(caller, "get_from"):
+                # supports custom error messages on individual containers
+                if location.db.get_from_err_msg:
+                    self.msg(location.db.get_from_err_msg)
+                else:
+                    self.msg("You can't get things from that.")
+                return
+
+        obj = caller.search(self.lhs, location=location)
+        if not obj:
+            return
+        if caller == obj:
+            self.msg("You can't get yourself.")
+            return
+        if not obj.access(caller, "get"):
+            if obj.db.get_err_msg:
+                self.msg(obj.db.get_err_msg)
+            else:
+                self.msg("You can't get that.")
+            return
+
+        # calling possible at_pre_get_from hook on location
+        if hasattr(location, "at_pre_get_from") and not location.at_pre_get_from(caller, obj):
+            return
```

Comment:
This hook has no default error message, so this failure will be completely silent? It seems like this could be confusing.